Libraries/Microsoft.Teams.Apps/Activities/Events/EventActivity.cs

```diff
@@ -1,23 +1,19 @@
 using Microsoft.Teams.Api.Activities;
+using Microsoft.Teams.Apps.Routing;
 
-namespace Microsoft.Teams.Apps.Routing;
+namespace Microsoft.Teams.Apps.Activities.Events;
 
 [AttributeUsage(AttributeTargets.Method, Inherited = true)]
 public class EventAttribute() : ActivityAttribute(ActivityType.Event, typeof(EventActivity))
 {
     public override object Coerce(IContext<IActivity> context) => context.ToActivityType<EventActivity>();
 }
 
-public partial interface IRoutingModule
+public static partial class AppEventActivityExtensions
 {
-    public IRoutingModule OnEvent(Func<IContext<EventActivity>, Task<object?>> handler);
-}
-
-public partial class RoutingModule : IRoutingModule
-{
-    public IRoutingModule OnEvent(Func<IContext<EventActivity>, Task<object?>> handler)
+    public static App OnEvent(this App app, Func<IContext<EventActivity>, Task<object?>> handler)
     {
-        Router.Register(new Route()
+        app.Router.Register(new Route()
```

Comment:
Help me understand this - so in typescript, events are emitted, but activities are routed. The main reason for this (i think) are that for events, we have more of a fire-and-forget system, whereas activities use a router and they may have multiple handlers that could get routed in order, and we might care about the result of the handlers to send back to the backend. 

In dotnet, it looks like everything is using the router

Reply:
its the same in .net, the events are just emitted, whats confusing you is that this is the event actitivty which is an activity with type `event`.